syntax = "proto3";

option go_package = ".;proto";

import "google/protobuf/empty.proto";

service Favorites {
    // 收藏
    rpc QueryFav(UserFavRequest)returns(FavResponse);
    rpc QueryFavList(UserFavRequest)returns(FavListResponse);
    rpc AddFav(UserFavRequest)returns(google.protobuf.Empty);
    rpc DeleteFav(UserFavRequest)returns(google.protobuf.Empty);
}

message UserFavRequest {
    int32 user_id = 1;
    int32 goods_id = 2;
}
message FavResponse {
    int32 user_id = 1;
    int32 goods_id = 2;
    string title = 3;
    float shop_price = 4;
    int32 stocks = 5;
    bool on_sale = 6;
}

message FavListResponse{
    int32 total = 1;
    repeated FavResponse data = 2;
}

message FavStatusResponse{
    bool checked = 1;
}
